### PR TITLE
SAK-44341 - escape the filename before building the resourceId for th…

### PR DESCRIPTION
**Pull request by maurercw**
…e check to see if it exists or not

### CODE REVIEW

**Review by bjones86 — APPROVED** [review]

File: content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesHelperAction.java [hunk 1]
```diff
@@ -2067,7 +2067,7 @@ private synchronized void doDragDropUpload(HttpServletRequest request, HttpServl
 				if (collection!=null)
 				{
 					//get the resourceId by using collectionName and uploadFileName
-					resourceId = collectionName +"/"+ uploadFileName;
+					resourceId = collectionName +"/"+ Validator.escapeResourceName(uploadFileName);
```
Comment (ern) [inline content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesHelperAction.java:2070]: Validator is deprecated try using FormattedText

File: content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesHelperAction.java [hunk 2]
```diff
@@ -2093,7 +2093,7 @@ private synchronized void doDragDropUpload(HttpServletRequest request, HttpServl
 					//So I disable this method call, though it can be enabled again if desired.
 					
 					//String resourceName = getUniqueFileName(uploadFileName, resourceGroup);
-					resourceId = resourceGroup + uploadFileName;
+					resourceId = resourceGroup + Validator.escapeResourceName(uploadFileName);
```
Comment (ern) [inline content/content-tool/tool/src/java/org/sakaiproject/content/tool/ResourcesHelperAction.java:2096]: Validator is deprecated try using FormattedText

**Review by ern — CHANGES_REQUESTED** [review]

**Comment by maurercw** [comment]
@ern I used Validator since that's already used (heavily) in that class.  I wanted everything to work exactly the same way as when it first creates the file.

**Comment by maurercw** [comment]
Looking into it a bit, there's no suitable replacement for escapeResourceName.  I'd also imagine that the tool would be super wonky if a different escaping mechanism was used down the road after resources were already created under the current mechanism.  You need to be able to rebuild the resourceId in a reliable way.  So, my vote is to continue with the status quo of the deprecated method.  Plus, it's only been deprecated since December 2011.  ;)

**Comment by maurercw** [comment]
I suppose I could move the escapeResourceName method over to the FormattedText service, but that seems like it's out of scope for this change as other tools that use it would be impacted.  If there was a suitable method already there, I'd happily use it.

**Review by ern — APPROVED** [review]

**Pull request merged**